### PR TITLE
Voice Verify Button Instead of Voice Verify Command

### PR DESCRIPTION
**Pull request by MrHemlock**
# Summary
Users have been having some issues knowing exactly what to do when trying to verify for voice permissions.  I've gone ahead and converted the command (which people never quite seemed to understand where to use the command) into an easy to press button.

I've also reworked how message deletions are handled within the cog, as the `delete_after` argument on sent messages so that we aren't handling it solely inside the cog.

The only thing that would need to be done before we pull the trigger on this would be to write up a quick message to replace the current one in the voice verify channel on the server.  Just to tweak to explain to use the button instead of the command

### CODE REVIEW

File: bot/exts/moderation/voice_gate.py
Comment (shtlrs) [reply bot/exts/moderation/voice_gate.py:222]: Also, the `text` part doesn't change AFAIK since it'll always have the same rules. So how about remove the burden of writing it out in Discord and have it saved as a constant ?

File: bot/exts/moderation/voice_gate.py
Comment (shtlrs) [inline bot/exts/moderation/voice_gate.py:83]: Moving this up first will help avoid hitting site if a user has the role already.

File: bot/exts/moderation/voice_gate.py
Comment (shtlrs) [inline bot/exts/moderation/voice_gate.py:89]: I think the button's callback is way too loaded with different responsibilities, it could help define the different "sub-boundaries" and checks better by factoring these out into separate private methods of the view.

That way it'll become easier to read & understand, and potentially refactor.

**Review by shtlrs — CHANGES_REQUESTED** [review]
Haven't tested this fully, but looks good  so far, thanks !

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:117]: I hadn't planned on removing the first join DM, but the more I think about it, message requests have made it harder to get messages to users anyway.  Yeah, I'll go ahead and drop it, good suggestion.

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:222]: The wording may change and other things may be added in the future.  Just seemed convenient at the time.

Comment (shtlrs) [reply bot/exts/moderation/voice_gate.py:222]: I've always thought it was just like the incidents channel, where the message is sent by a regular user rather than a bot user.

That way we get rid of these constrains.

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:63]: I'm not sure on this one.  It feels less explicit to me.

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:89]: Fair point, I'll see how I can chop it up.

File: bot/exts/moderation/voice_gate.py
Comment (Robin5605) [reply bot/exts/moderation/voice_gate.py:222]: I think in incidents it's fine since we just have reactions, but we can't add a button to a user-sent message, so this unfortunately has to be sent by the bot

Comment (Robin5605) [reply bot/exts/moderation/voice_gate.py:222]: I think what we could do is if the user invoking this command omits the channel, then the default channel would be `#voice-verify`. If they specify a channel, the message would go there. 

Same deal with the text, maybe we could set the fallback text value to what we already have, but it can be overridden if necessary.

@MrHemlock I think we'll also either need to delete the original message that invoked the command if it's sent in `#voice-verify` so the initial command message doesn't linger

Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:222]: I was thinking we would be doing it in a channel like mod-spam or admin-spam.  And we'd have the ability to just delete the message itself.  Not only that, not deleting it right away makes it easy to re-copy and fix if I (inevitably) type something incorrectly while putting up the rules and message.  I feel like this command really doesn't need to have a lot of functionality tied up with it.  It's only going to be used once in a blue moon, likely just by me.

And yes, because it's adding a view, it has to come from a bot.

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:48]: Changed

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:89]: After looking this over multiple times, I just don't see the benefit of splitting it up.  The methods I would be creating would mostly be one line returns.  It feels like it ends up more.... I guess Java like, with gets that don't really benefit the code.  

I do see where it could help readability by being able to just look at the general class structure and see the method names, but at the same time, splitting it up leads to a couple complications.  For instance, for the try block, I considered making a `_get_metricity_data()` method.  But then would I have it handle the exceptions and the messages those send?  If I did, then it'd be a method that'd be doing more than its namesake and if I didn't, I'd be making a method that's adding yet another call on top of already just `send`ing the message.  

I kind of think splitting it up would hurt readability a bit, or just add method call chains that don't really need to be there.

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py]: Scooted up

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py]: Removed

**Review by shtlrs — CHANGES_REQUESTED** [review]
Just a couple more things to improve and we should be good, thanks !

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:75]: Ah right, totally forgot that.  That always throws me off.

File: bot/exts/moderation/voice_gate.py
Comment (MrHemlock) [reply bot/exts/moderation/voice_gate.py:160]: Oh huh, didn't know we had that function.  For sure going to apply that one.

**Comment by MrHemlock** [comment]
Re-adding the "review: do not merge" label as someone will need to be around to send the new view message to the voice channel.

**Review by shtlrs — APPROVED** [review]
Looking good, thanks !

File: bot/exts/moderation/voice_gate.py
Comment (wookie184) [inline bot/exts/moderation/voice_gate.py:37]:
```suggestion
    "Click the Voice Verify button above to verify. "
```
May be slightly clearer

**Review by wookie184 — APPROVED** [review]
Great stuff, i'll let you merge in case you want to apply my suggestion first.

**Pull request merged**
